File: charfreq 6 Kyu.js
function letterFrequency(text){

    var hashTable = {}; //O(1)
    var outputList = []; //O(1)

    var cleanString = text.replace(/[^A-Za-z]+/g, '').toLowerCase() //O(3)

    for (eachLetter of cleanString){ //O(n)
      (!(hashTable.hasOwnProperty(eachLetter))) ? hashTable[eachLetter] = 1 : hashTable[eachLetter] ++
      }

    for (key in hashTable){ //O(n)
      var pair = []
      pair.push(key);
      pair.push(parseInt(hashTable[key]))
      outputList.push(pair) //O(2)
    }

    console.log(outputList)
    var arr = outputList.sort(function(a,b) {
      if (a[1] === b[1]){
        return (a > b) ? 1 : -1;
      }
      return b[1] - a[1]
      });
    console.log(arr);
  };
letterFrequency("As long as I'm learning something, I figure I'm OK - it's a decent day.")

File: charfreq 6 Kyu.py
import re
def letter_frequency(text):

    hashTable = {}
    cleanString = re.sub("[^A-Za-z]+", "", text).lower()

    for eachLetter in cleanString:
        hashTable[eachLetter] = hashTable.get(eachLetter, 0) + 1

    listByAlpha = sorted(hashTable.items(), key=lambda x: x[0])
    outputList = sorted(listByAlpha, key=lambda x: x[1], reverse=True)
    print outputList

letter_frequency('JJJKKKbb')
